### PR TITLE
Move ML tests to be sub-projects of ML

### PR DESCRIPTION
**Pull request by jasontedor**
This commit moves the ML QA tests to be a sub-project of ML. The purpose of this refactoring is to enable ML developers to run :x-pack:plugin:ml:check and run the vast majority of a ML tests with a single command (this still does not contain the ML REST tests, nor the upgrade tests). This simplifies local development for faster iteration.

### CODE REVIEW

**Comment by jasontedor** [comment]
Note: this is similar to the approach that we are taking in CCR and ILM.

**Comment by davidkyle** [comment]
@droberts195 is on holiday I've pinged some other ml'ers for reivew

File: x-pack/plugin/ml/qa/build.gradle [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import org.elasticsearch.gradle.test.RestIntegTestTask
+
+subprojects {
+    // HACK: please fix this
+    // we want to add the rest api specs for xpack to qa tests, but we
+    // need to wait until after the project is evaluated to only apply
+    // to those that rest tests. this used to be done automatically
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/qa/build.gradle]: nit: `to those that tests`: Remove "that"?

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM This is awesome Jason! Left a small typo but you can merge with or without fixing it.

**Pull request merged**